File: E-Booking/E-Booking.Models/SpecialistTransportRequestModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;

namespace EBooking.Models
{
    public class SpecialistTransportRequestModel
    {
        public bool IsHandledByProfessional { get; set; }

        public bool IsWhilstLayingDown { get; set; }

        public bool IsOxygenTheropy { get; set; }

        public bool IsPrecludesTravelling { get; set; }

        public bool IsAdmission { get; set; }

        public bool IsVisitOrAdmitted { get; set; }

        public bool IsVisit { get; set; }

        public string AuthorisingConsultantOrGP { get; set; }

        public string AuthorisingPracticeName { get; set; }
    }
}

File: E-Booking/E-Booking/Scripts/KnockoutModels/SpecialistTransportRequestModel.js
function SpecialistTransportRequestModel()
{
    var self = this;

    self.IsHandledByProfessional = ko.observable(false);
    self.IsWhilstLayingDown = ko.observable(false);
    self.IsOxygenTheropy = ko.observable(false);
    self.IsPrecludesTravelling = ko.observable(false);

    self.IsAdmission = ko.observable(false);
    self.IsVisitOrAdmitted = ko.observable(false);
    self.IsVisit = ko.observable(false);

    self.AuthorisingConsultantOrGP = ko.observable().extend({ required: true });
    self.AuthorisingPracticeName = ko.observable().extend({ required: true });
    
    self.IsHandledByProfessional.extend({ checked: { onlyIf: function () { return checkedAtLeastReasonOneAnswerSelected(self); } } });
    self.IsWhilstLayingDown.extend({ checked: { onlyIf: function () { return checkedAtLeastReasonOneAnswerSelected(self); } } });
    self.IsOxygenTheropy.extend({ checked: { onlyIf: function () { return checkedAtLeastReasonOneAnswerSelected(self); } } });
    self.IsPrecludesTravelling.extend({ checked: { onlyIf: function () { return checkedAtLeastReasonOneAnswerSelected(self); } } });

    self.IsAdmission.extend({ checked: { onlyIf: function () { return checkedAtLeastReasonTwoAnswerSelected(self); } } });
    self.IsVisitOrAdmitted.extend({ checked: { onlyIf: function () { return checkedAtLeastReasonTwoAnswerSelected(self); } } });
    self.IsVisit.extend({ checked: { onlyIf: function () { return checkedAtLeastReasonTwoAnswerSelected(self); } } });
    
    self.errors = ko.validation.group(self);
}

function checkedAtLeastReasonOneAnswerSelected(self) {
    return !(self.IsHandledByProfessional() || self.IsWhilstLayingDown() || self.IsOxygenTheropy() || self.IsPrecludesTravelling());
}
function checkedAtLeastReasonTwoAnswerSelected(self) {
    return !(self.IsAdmission() || self.IsVisitOrAdmitted() || self.IsVisit());
}